### PR TITLE
[C#] add indentation tests, always unindent if closing brace begins line

### LINKED ISSUE #3653: [C#] Unpleasing auto indent behavior
### What happened?

Reported from discord: https://discord.com/channels/280102180189634562/280157083356233728/1071057792288440320

> is there a setting I can change that will stop ST from indenting the closing bracket line while I'm typing the `else` as seen in the video?

https://user-images.githubusercontent.com/6594915/216628428-b699e226-be5d-4ea6-9627-128f52c1ca60.mp4

I've tested that JS/PHP/C/C++ won't behave like this.

### PR DESCRIPTION
**Pull request by keith-hall**
[C#] add indentation tests, always unindent if closing brace is the first non-whitespace character on the line, regardless of what follows it

Fixes #3653

### CODE REVIEW

File: C#/tests/syntax_test_indentation.cs [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+    {
+        get => lastName;
+        init => lastName = (value ?? throw new ArgumentNullException(nameof(LastName)));
+    }
+}
+
+void Main ()
+{
+    var otherPerson = person with { LastName = "Hanselman" };
+
```
Comment (jfcherng) [inline C#/tests/syntax_test_indentation.cs:56]: is this empty line intentional?

Comment (keith-hall) [reply C#/tests/syntax_test_indentation.cs:56]: it was, but probably doesn't serve much purpose thinking about it :)

**Review by jfcherng — APPROVED** [review]

**Review by deathaxe — APPROVED** [review]
